fs/operations/operations.go

```diff
@@ -343,6 +344,112 @@ func CommonHash(fa, fb fs.Info) (hash.Type, *fs.HashesOption) {
 // It returns the destination object if possible.  Note that this may
 // be nil.
 func Copy(ctx context.Context, f fs.Fs, dst fs.Object, remote string, src fs.Object) (newDst fs.Object, err error) {
+	resumeOpt := &fs.OptionResume{}
+	if doResume := f.Features().Resume; doResume != nil {
+		// Creates an OptionResume that will be passed to Put/Upload
+		resumeOpt = &fs.OptionResume{ID: "", Pos: 0, SetID:
+		// SetID will be called by backend's Put/Update function if the object's upload
+		// could be resumed upon failure
+		//
+		// SetID takes the passed resume ID and Fingerprint of the object and stores it in
+		// --cache-dir so that future Copy operations can resume the upload if it fails
+		func(ID string) (err error) {
```

Comment:
This should be a separate function too I think, so the closure you pass as SetID: is only a line or two long.